.github/workflows/csharp.yml

```diff
@@ -52,35 +56,40 @@ jobs:
       EVENT_TYPE: ${{ github.event_name }}
       CURRENT_BRANCH: ${{ github.head_ref || github.ref_name }}
     steps:
-      - uses: actions/checkout@v3
-      - name: Setup .NET
-        uses: actions/setup-dotnet@v1
-        with:
-          dotnet-version: 5.0.x
-      - name: Restore nuget packages
-        run: |
-          nuget restore OptimizelySDK.Travis.sln
-          nuget install ./OptimizelySDK.Tests/packages.config -OutputDirectory ./packages
-          nuget install NUnit.Runners -Version 2.6.4 -OutputDirectory ./testrunner
-      - name: script
-        run: |
-          ./install_mono.sh
-          xbuild /p:SignAssembly=true /p:AssemblyOriginatorKeyFile=$(pwd)/keypair.snk /p:Configuration=Release ./OptimizelySDK.Travis.sln
-          mono ./testrunner/NUnit.Runners.2.6.4/tools/nunit-console.exe ./OptimizelySDK.Tests/bin/Release/OptimizelySDK.Tests.dll
-      - name: find and sign dll
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v1
+      - name: Setup NuGet
+        uses: NuGet/setup-nuget@v1
+      - name: Restore NuGet packages
+        run: nuget restore ./OptimizelySDK.Travis.sln
+      - name: Build solution
+        run: msbuild /p:SignAssembly=true /p:AssemblyOriginatorKeyFile=$(pwd)/keypair.snk /p:Configuration=Release ./OptimizelySDK.Travis.sln
+      - name: Install NUnit Console
+        run: nuget install NUnit.Console -Version 3.15.2 -DirectDownload -OutputDirectory .
+      - name: Run NUnit tests
+        run: ./NUnit.ConsoleRunner.3.15.2\tools\nunit3-console.exe /timeout 10000 /process Separate ./OptimizelySDK.Tests/bin/Release/OptimizelySDK.Tests.dll
+      - name: Find and sign all DLLs
         id: unit_tests
         run: |
-          sudo find . -path './OptimizelySDK*bin/Release/OptimizelySDK*.dll' -not -regex '.*Tests.*' -print0 | while IFS= read -r -d '' file; do sn -R $file ./keypair.snk; done
-      - name: Check on success
+          $sn = "C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\sn.exe";
```

Comment:
great job. can you please find a way if there is any variable to represent base path. don't hardcode NETFX4.8 path.

Reply:
Thanks.

I struggled to enumerate the path to copy of `sn.exe`.  There are several copies of it on the `windows-2019` image.  The PowerShell to find and select only one is not clear to the reader, so I decided to hard-code the path.  

I believe leaving this hard-coded poses little risk. The Microsoft's image *should* (in theory) not change since it's shared among so many builds.

I do think it's a great idea to move this base path to a YAML level variable, so it's prominent and if needed can be updated.